### PR TITLE
fix PlayerItemHeldEvent firing twice

### LINKED ISSUE #5482: PlayerItemHeldEvent fires twice
### Expected behavior

When the event is canceled it is not executed again.

### Observed/Actual behavior

The event is executed again after canceling a PlayerItemHeldEvent.

### Steps/models to reproduce

1. Cancel the ``PlayerItemHeldEvent`` when it is executed and print a test message to the console.
2. Observe

### Plugin list

None, edited the server directly.

### Paper version

This server is running Paper version git-Paper-54555252a (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
Error obtaining version information
Previous version: git-Paper-c310f0a61 (MC: 1.16.5)

### Agreements

I am running the latest version of Paper available from https://papermc.io/downloads.
I have searched for and ensured there isn't already an open issue regarding this.
My version of Minecraft is supported by Paper.

### PR DESCRIPTION
**Pull request by Machine-Maker**
Fixes #5482

### CODE REVIEW

File: Spigot-Server-Patches/0707-fix-PlayerItemHeldEvent-firing-twice.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
```
Comment (Proximyst) [inline Spigot-Server-Patches/0707-fix-PlayerItemHeldEvent-firing-twice.patch]: Author to chickeneer

**Review by Proximyst — APPROVED** [review]
LGTM. One note for merger.

**Comment by electronicboy** [comment]
I'm like, 90% brain dead right now, but, this should maybe be merged into the patch which I think caused this, the fixing of something else?

**Comment by kennytv** [comment]
This is an upstream issue, there's no patch this could be merged into

**Comment by electronicboy** [comment]
ah, nvm then, thought this was caused by the last patch

**Review by electronicboy — APPROVED** [review]

**Pull request merged**